docs/recipes/ComputingDerivedData.md

```diff
@@ -130,6 +130,194 @@ const VisibleTodoList = connect(
+
+#### `selectors/todoSelectors.js`
+
+```js
+import { createSelector } from 'reselect'
+
+const getVisibilityFilter = (state, props) => state.todoLists[props.listId].visibilityFilter
+const getTodos = (state, props) => state.todoLists[props.listId].todos
+
+const getVisibleTodos = createSelector(
+  [ getVisibilityFilter, getTodos ],
+  (visibilityFilter, todos) => {
+    switch (visibilityFilter) {
+      case 'SHOW_COMPLETED':
+        return todos.filter(todo => todo.completed)
+      case 'SHOW_ACTIVE':
+        return todos.filter(todo => !todo.completed)
+      default:
+        return todos
+    }
+  }
+)
+
+export default getVisibleTodos
+```
+
+`props` can be passed to `getVisibleTodos` from `mapStateToProps`:
+
+```js
+const mapStateToProps = (state, props) => {
+  return {
+    todos: getVisibleTodos(state, props)
+  }
+}
+```
+
+So now `getVisibleTodos` has access to `props`, and everything seems to be working fine.
+
+**But there is a problem!**
+
+Using the `getVisibleTodos` selector with multiple instances of the `visibleTodoList` container will not correctly memoize:
+
+#### `containers/VisibleTodoList.js`
+
+```js
+import { connect } from 'react-redux'
+import { toggleTodo } from '../actions'
+import TodoList from '../components/TodoList'
+import { getVisibleTodos } from '../selectors'
+
+const mapStateToProps = (state, props) => {
+  return {
+    // WARNING: THE FOLLOWING SELECTOR DOES NOT CORRECTLY MEMOIZE
+    todos: getVisibleTodos(state, props)
+  }
+}
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    onTodoClick: (id) => {
+      dispatch(toggleTodo(id))
+    }
+  }
+}
+
+const VisibleTodoList = connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(TodoList)
+
+export default VisibleTodoList
+```
+
+A selector created with `createSelector` only returns the cached value when its set of arguments is the same as its previous set of arguments. If we alternate between rendering `<VisibleTodoList listId="1" />` and `<VisibleTodoList listId="2" />`, the shared selector will alternate between receiving `{listId: 1}` and `{listId: 2}` as its `props` argument. This will cause the arguments to be different on each call, so the selector will always recompute instead of returning the cached value. We'll see how to overcome this limitation in the next section.
+
+### Sharing Selectors Across Multiple Components
+
+> The examples in this section require React Redux v4.3.0 or greater
+
+In order to share a selector across multiple `VisibleTodoList` components **and** retain memoization, each instance of the component needs its own private copy of the selector.
+
+Let's create a function named `makeGetVisibleTodos` that returns a new copy of the `getVisibleTodos` selector each time it is called:
+
+#### `selectors/todoSelectors.js`
+
+```js
+import { createSelector } from 'reselect'
+
+const getVisibilityFilter = (state, props) => state.todoLists[props.listId].visibilityFilter
```

Comment:
Let’s put some line breaks here, it’s getting too long.
```js
const getVisibilityFilter = (state, props) =>
  state.todoLists[props.listId].visibilityFilter
```

same for `getTodos`